frontend/src/components/categoryList/categoryList.js

```diff
@@ -1,24 +1,110 @@
+            showArray[index] = false;
+        }
+        show.current = showArray;
+        expand.current = expandArray;
+        setRefresh(r => r + 1);
+    }
+
     useEffect(() => {
-        categoryService.getAll()
+        categoryService.getCategoriesWithSubcategories()
             .then((response) => {
                 setCategories(response.data);
-            });
-    }, []);
+                if (filter) {
+                    prodCat.current = "PRODUCT ";
+                    if(sign === -1){
+                         show.current = new Array(response.data.length).fill(false);
+                        expand.current = new Array(response.data.length).fill(plus);
+                    }
+                    let selected = response.data.find(element => element.category.name.toLowerCase() === filter.toLowerCase());
+                    let showArray = [...show.current];
+                    if (selected) {
+                        if (previousCategory.current !== -1 && previousCategory.current !== selected.category.id - 1) {
+                            showArray[previousCategory.current] = false;
+                            expand.current[previousCategory.current] = plus;
+                        }
+                        showArray[selected.category.id - 1] = true;
+                        show.current = showArray;
+                        if (previousCategory.current !== -1) {
+                            expand.current[selected.category.id - 1] = minus;
+                        } else setSign(1);
+                        previousCategory.current = selected.category.id - 1;
+
+                    } else {
+                        if (previousCategory.current !== -1) {
+                            showArray[previousCategory.current] = false;
+                            expand.current[previousCategory.current] = plus;
+                        }
+                    }
+                }
+            })
+        if (sign === 1) {
+            expand.current[previousCategory.current] = minus;
+        }
+
+    }, [filter, sign, refresh]);
+    if (filter) {
+        mainContainerClass += " filter";
+        listContainerClass += "filter";
+    }
     return (
-        <div className="list">
-            <p>CATEGORIES</p>
+        <div className={mainContainerClass}>
+            <p>{prodCat.current}CATEGORIES</p>
             <ul className='category'>
                 {categories && categories.map(cat => (
-                    <li key={cat.id}>{cat.name}</li>
+                    <div key={cat.category.id}>
+                        <div className={listContainerClass}>
+                            <Link to={`/shop/${cat.category.name}`}>
+                                <li>{cat.category.name} </li>
+                            </Link>
+                            {filter &&
+                            <img id={cat.category.id} className="showMore" src={expand.current[cat.category.id-1]} onClick={showSubcategories}
```

Comment:
Is it possible to add here that event.target.id or src, whatever you need like:
onClick={event => showSubcategories(event.target.src)}